### DISCUSSION TITLE
Is aws-sdk-s3 client API cancellaction safe?

### DISCUSSION THREAD

**Discussion by MrCroxx** [awslabs/aws-sdk-rust#994] [Q&A]
Hi, I'm using `aws-sdk-s3` in a cloud-native database project. I use `tokio` as the async runtime in Rust. I wonder if `aws-sdk-s3` APIs are *cancellation safe*? In some cases, the async task that called `aws-sdk-s3` API can be cancelled. Will that causes deadlocks or undefined behaviors?

**Comment by rcoh** [comment]
Yes, in principle (and it aims to be). However, it's not a behavior we have extensively tested. Generally, however, the number of resources that are shared between subsequent requests is fairly small and that code (e.g. the identity provider cache) is very carefully tested.
